util/esp32_wifi.c

```diff
@@ -1302,44 +1401,85 @@ CgiStatus tplWlan(HttpdConnData *connData, char *token, void **arg)
             snprintf(buff, sizeof(buff) - 1,
-                         "<b>Can't scan in this mode.</b><br> "
-                         "Click <a href=\"setmode.cgi?mode=%d\">here</a> "
-                         "to go to STA+AP mode.<br>", WIFI_MODE_APSTA);
+                    "<p><button onclick=\"location.href='setmode.cgi?mode=%d'\">Go to <b>AP+STA</b> mode</button> (Both AP and Client)</p>",
+                    WIFI_MODE_APSTA);
             break;
         case WIFI_MODE_APSTA:
             snprintf(buff, sizeof(buff) - 1,
-                     "Click <a href=\"setmode.cgi?mode=%d\">here</a> "
-                     "to go to standalone AP mode.<br>", WIFI_MODE_AP);
+                    "<p><button onclick=\"location.href='setmode.cgi?mode=%d'\">Go to standalone <b>AP</b> mode</button> (Access Point Only)</p>",
+                    WIFI_MODE_AP);
 
             /* Only offer switching to STA mode if we have a connection. */
             if(sta_connected()){
                 snprintf(buff + strlen(buff), sizeof(buff) - strlen(buff) - 1,
-                         "Click <a href=\"setmode.cgi?mode=%d\">here</a> "
-                         "to go to STA mode.<br>", WIFI_MODE_STA);
+                        "<p><button onclick=\"location.href='setmode.cgi?mode=%d'\">Go to standalone <b>STA</b> mode</button> (Client Only)</p>",
+                        WIFI_MODE_STA);
             }
             break;
         case WIFI_MODE_STA:
         default:
             snprintf(buff, sizeof(buff) - 1,
-                     "Click <a href=\"setmode.cgi?mode=%d\">here</a> "
-                     "to go to standalone AP mode.<br>"
-                     "Click <a href=\"setmode.cgi?mode=%d\">here</a> "
-                     "to go to AP+STA mode.<br>",
-                     WIFI_MODE_AP, WIFI_MODE_APSTA);
+                    "<p><button onclick=\"location.href='setmode.cgi?mode=%d'\">Go to standalone <b>AP</b> mode</button> (Access Point Only)</p>"
+                    "<p><button onclick=\"location.href='setmode.cgi?mode=%d'\">Go to <b>AP+STA</b> mode</button> (Both AP and Client)</p>",
+                    WIFI_MODE_AP, WIFI_MODE_APSTA);
             break;
         }
 
         /* Always offer WPS. */
-        strlcat(buff, "Click <a href=\"startwps.cgi\">here</a> "
-                      "to connect to AP with WPS. This will switch to "
-                      "AP+STA mode. You can switch to STA only mode "
-                      "after the client has connected.<br>",
-                      sizeof(buff) - strlen(buff));
+        snprintf(buff + strlen(buff), sizeof(buff) - strlen(buff) - 1,
+                "<p><button onclick=\"location.href='startwps.cgi'\">Connect to AP with <b>WPS</b></button> "
+                "This will switch to AP+STA mode. You can switch to STA only mode "
+                "after the client has connected.</p>");
+
+        /* Disable WiFi.  (only available if Eth connected?) */
+        if(1){//eth_connected()){
```

Comment:
Were you planning to change this in the future to detect ethernet?

Reply:
No, I just left it there as an example.  I guess some applications would not want to provide a way to completely disable access.  
But my application needs it.
I'll just remove the if(1) junk if you want me to.